pkg/api/artifact.go

```diff
@@ -1,9 +1,29 @@
 package api
 
 import (
+	"context"
 	"fmt"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/containerdisks/pkg/docs"
 )
 
+type ArtifactTest func(ctx context.Context, vmi *v1.VirtualMachineInstance, params *ArtifactTestParams) error
+
+type ArtifactTestParams struct {
+	// Username is the username used to login into the VM
+	Username string
+	// PrivateKey is the private key used to login into the VM
+	PrivateKey interface{}
+}
+
+type ArtifactResult struct {
```

Comment:
Since we kind of build a pipeline with the new commands, we probably want to add an indicator if something should have been created/verified but failed. We can do that in a follow up.